mod/state-transition/pkg/core/state/statedb.go

```diff
@@ -283,21 +293,30 @@ func (s *StateDB[
 			withdrawalIndex++
 		}
 
-		// Cap the number of withdrawals to the maximum allowed per payload.
-		//#nosec:G701 // won't overflow in practice.
-		if len(withdrawals) == int(s.cs.MaxWithdrawalsPerPayload()) {
-			break
-		}
-
 		// Increment the validator index to process the next validator.
-		validatorIndex = (validatorIndex + 1) % math.ValidatorIndex(
-			totalValidators,
-		)
+		validatorIndex++
+		validatorIndex %= math.ValidatorIndex(totalValidators)
 	}
 
 	return withdrawals, nil
 }
 
+// EVMInflationWithdrawal returns the withdrawal used for EVM balance inflation.
+//
+// NOTE: The withdrawal index and validator index are both set to 0 as they are
+// not used during processing.
+func (s *StateDB[
+	_, _, _, _, _, _, _, _, WithdrawalT, _,
+]) EVMInflationWithdrawal() WithdrawalT {
+	var withdrawal WithdrawalT
+	return withdrawal.New(
+		0,
```

Comment:
the one problem I have with this choice is that 0 is a valid validator. Granted that validator index should not make a difference from the EL perspective, I still wonder if it's better, even just estetically, to pick the block proposer as validator to "withdraw minting to"

Reply:
I chose 0 to explicitly state its not for use downstream. We can create a uint64 constant `EVMInflationWithdrawalIndex` and `EVMInflationWithdrawalValidatorIndex` if something like that is more clear